### PR TITLE
Pure cn

### PR DESCRIPTION
**Pull request by 5im1z**
Add PureCN files to the WGS pipeline folder. Files generated by @colganwi

### CODE REVIEW

File: WGS_pipeline/PureCN_pipeline/PureCN.wdl [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    String wgd = wgd_table[0]
+    String loh_fraction = wgd_table[1]
+    String cin = wgd_table[2]
+    String cin_allele_specific = wgd_table[3]
+    String cin_ploidy_robust = wgd_table[4]
+    String cin_allele_specific_ploidy_robust = wgd_table[5]
+  }
+
+  runtime {
+    docker: "markusriester/purecn:latest"
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/PureCN.wdl:71]: never use latest

Comment (jkobject) [reply WGS_pipeline/PureCN_pipeline/PureCN.wdl:71]: let the version be a parameter

**Review by jkobject — COMMENTED** [review]
let's try to use our list of intervals at least for the agilent/ice ones

File: WGS_pipeline/PureCN_pipeline/call_wgd_and_cin.R [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#Function from PureCN
+callCIN <- function(loh,allele.specific = TRUE, reference.state =
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/call_wgd_and_cin.R:2]: Let's add a bit of documentation on the R scripts please.

File: WGS_pipeline/PureCN_pipeline/PureCN.wdl [hunk 3]
```diff
@@ -0,0 +1,112 @@
+task run_PureCN {
+
+  # File-related inputs
+  String sampleID
+  File segFile
+  File vcf
+  File intervals
+  File call_wgd_and_cin_script
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/PureCN.wdl]: use the git clone model instead for versioning

**Review by jkobject — COMMENTED** [review]
Just need to finish the SOP and add in the notebook (maybe it is already there?)

File: WGS_pipeline/PureCN_pipeline/PureCN.wdl (hunk 1 above)
Comment (5im1z) [reply WGS_pipeline/PureCN_pipeline/PureCN.wdl]: I fixed this on dev but forgot to update it here. done now!

File: WGS_pipeline/PureCN_pipeline/PureCN_update_solution.wdl [hunk 4]
```diff
@@ -0,0 +1,79 @@
+task run_update_solution {
```
[the remark this answers is not in the document]
Comment (colganwi) [reply WGS_pipeline/PureCN_pipeline/PureCN_update_solution.wdl:1]: This updates WGD, CIN, and other outputs after manual curation.

**Comment by jkobject** [comment]
Will merge once:
1. the git clone model is used
2. the .R scripts have some more documentation about the role of the functions
3. there is some link to an SOP for selecting the right solutions (right now, there is a description of everything but not a process to follow to solve issues).

File: PureCN_curation.ipynb [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e4bb6282",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wm = dm.WorkspaceManager(WORKSPACE)\n",
+    "samples_df = wm.get_samples()\n",
+    "#samples_df = pd.read_csv(\"~/Downloads/sample.tsv\",sep = \"\\t\")"
```
Comment (jkobject) [inline PureCN_curation.ipynb:62]: don't leave access to random files in here.

File: PureCN_curation.ipynb [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+   "source": [
+    "# Manually inspect and update solutions "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50baf673",
+   "metadata": {},
+   "source": [
+    "Now that all the solution PDFs and \"to_curate.csv\" are downloaded, open the solution PDFs locally and manually inspect them following the guidelines here: https://docs.google.com/document/d/1Rte0xKK3ZE_UV6MWepdXRIbAehUJg8FuLaDckrWhPTQ/edit?usp=sharing. Fill in the PureCN_curated_solution column for all samples in to_curate.csv with the index of the most reasonable solution. Note that solution indices are 1-based. If none of the solutions look right, set PureCN_failed = \"TRUE\" & PureCN_curated_solution = 0."
```
Comment (jkobject) [inline PureCN_curation.ipynb:145]: list in readme too.

File: PureCN_curation.ipynb [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26394795",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "curated = pd.read_csv(\"~/Desktop/to_curate.csv\",index_col = 0)\n",
```
Comment (jkobject) [inline PureCN_curation.ipynb:163]: same

File: PureCN_curation.ipynb (hunk 6 above)
Comment (5im1z) [reply PureCN_curation.ipynb:146]: I think the readme is a bigger problem (a lot more things are outdated besides mentions of PureCN) so we might want to update it as a whole after this merge?

File: WGS_pipeline/PureCN_pipeline/PureCN.wdl [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+		# Method configuration inputs
+		String genome="hg38"
+		Int maxCopyNumber=8
+		Float minPurity=0.90
+		Float maxPurity=0.99
+		String funSegmentation="Hclust"
+		Int maxSegments=1000
+		String otherArguments="--post-optimize --model-homozygous --min-total-counts 20"
+        String purecn_docker="markusriester/purecn:2.2.0"
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/PureCN.wdl:21]: issue with tab use

File: PureCN_curation.ipynb (hunk 6 above)
Comment (5im1z) [reply PureCN_curation.ipynb:146]: Turns out William had already linked it in the purecn readme!

File: WGS_pipeline/PureCN_pipeline/README.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+**WES ICE**
+
+`Rscript $PURECN/IntervalFile.R --in-file ice_hg38_lifted_chrXY.no_header.bed --fasta ~/Data/VCFs/Liftover/hg38.fa --out-file ice_hg38_intervals.txt --genome hg38 --mappability GCA_000001405.15_GRCh38_no_alt_analysis_set_100.bw`
+
+**WGS**
+
+Here we uniformly sample 2% of the genome to use for absolution copy number inference.
+
+`Rscript make_wgs_intervals.R`
+
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/README.md:24]:
```suggestion

We have also created new intervals from the recommended pureCN workflow, including mappability information on both the wgs intervals and wes intervals
```

File: WGS_pipeline/PureCN_pipeline/select_solution.R [hunk 10]
```diff
@@ -0,0 +1,246 @@
+library(VariantAnnotation)
+
```
Comment (jkobject) [inline WGS_pipeline/PureCN_pipeline/select_solution.R:2]:
```suggestion
####################### 

This is a copy of a piece of code from PureCN to update results and plots from a solution to the duplication value.

#############
```

**Pull request merged**